src/include/catalog/schema.h

```diff
@@ -52,17 +59,20 @@ class Schema {
      * @param max_varlen_size the maximum length of the varlen entry
      * @param nullable true if the column is nullable, false otherwise
      * @param oid internal unique identifier for this column
+     * @param default_value default value for this column. Null by default
      */
     Column(std::string name, const type::TypeId type, const uint16_t max_varlen_size, const bool nullable,
-           const col_oid_t oid)
+           const col_oid_t oid, byte *default_value = nullptr)
         : name_(std::move(name)),
           type_(type),
           attr_size_(type::TypeUtil::GetTypeSize(type_)),
           max_varlen_size_(max_varlen_size),
           nullable_(nullable),
           oid_(oid) {
+      // TODO(Sai): How to handle default values for VARLEN?
       TERRIER_ASSERT(attr_size_ == VARLEN_COLUMN, "This constructor is meant for VARLEN columns.");
       TERRIER_ASSERT(type_ != type::TypeId::INVALID, "Attribute type cannot be INVALID.");
+      SetDefault(default_value);
```

Comment:
This line is problematic. The most significant bit of VARLEN_COLUMN is set to 1. You should not use that directly as the size for `memcpy` in `SetDefault`, which will cause scary memory issues.